lib/lurker/templates/lurker/rendering/_submit_form.html.erb

```diff
@@ -6,100 +6,145 @@
-      state.curl = this.renderCurl(method, host, template, values, jQuery.param(JSON.parse('<%= @post_params.to_json.html_safe %>')));
       return state;
     },
-    getPayload: function() {
-      // FIXME
-      return jQuery('#post-payload input').serialize();
-    },
-    getHostname: function(host) {
-      return host;
-    },
-    renderAction: function(host, template, values) {
-      var action = template;
-      var i;
-      for (i in values) {
+    renderAction: function() {
+      var action = this.state.template;
+      var values = this.state.values;
+      for (var i = 0; i < values.length; i++) {
         action = action.replace(new RegExp(':' + values[i].label), values[i].value);
       }
-      return this.getHostname(host) + action;
+      return this.state.host + action;
     },
-    renderCurl: function(method, host, template, values, payload) {
-      var url = '"' + this.renderAction(host, template, values) + '"';
+    renderPayload: function() {
+      var contentType = this.state.requestMediaType;
+      if (contentType === 'application/json' || /\+json$/.test(contentType)) {
+        return JSON.stringify(this.state.payload);
+      } else {
+        // default to 'application/x-www-form-urlencoded'
+        return jQuery.param(this.state.payload);
+      }
+    },
+    renderCurl: function() {
+      var url = this.renderAction();
+      var payload = this.renderPayload();
+
+      var results = [];
+      results.push('-X ' + this.state.method);
       if (payload.length > 0) {
-        url = '-d "' + (payload || this.getPayload()) + '" ' + url;
+        results.push('-H "Content-Type: ' + this.state.requestMediaType + '"');
+        results.push('-d "' + payload.replace(/"/g, '\\"') + '"');
       }
-      return 'curl -X ' + method + " " + url;
+      return 'curl ' + (results.join(' ')) + ' "' + url + '"';
     },
-    // see react-catalyst.js
-    afterSetPartialState: function() {
-      this.setState({
-        action: this.renderAction(this.state.host, this.state.template, this.state.values),
-        curl: this.renderCurl(this.state.method, this.state.host, this.state.template, this.state.values, jQuery('#post-payload input').serialize())
-      });
+    renderHostOptions: function() {
+      var hostOptions = [];
+      for (var name in this.state.hosts) {
+        hostOptions.push(<option key={name} value={this.state.hosts[name]}>{name}</option>);
+      }
+      return hostOptions;
+    },
+    renderRequestMediaTypeOptions: function () {
+      var mediaTypes = this.state.requestMediaTypes;
+      var requestMediaTypeOptions = [];
+      for (var i = 0; i < mediaTypes.length; i++) {
+        requestMediaTypeOptions.push(<option key={i}>{mediaTypes[i]}</option>)
+      }
+      return requestMediaTypeOptions;
     },
     handleSubmit: function() {
-      // FIXME
-      Lurker.onSubmit(jQuery('#payload'));
+      Lurker.disableSubmitButton();
```

Comment:
yes, the code certainly belongs to the component, but nowadays `Lurker` component in js it all about this form, as well. 
I think we cannot eliminate passing variables from ruby to js and these 2 parts will continue to live side-by-side. 
we probably can inline all the `detectContentType`, `fillInInfoTab` into here, but I really dislike the idea about js embedded in erb. as such I suggest:

- extract https://github.com/nebolsin/lurker/blob/feature/json-request-format/lib/lurker/templates/javascripts/lurker.js#L3L43 to say `lurker-form.js` and include it into sprockets manifest
- put onSubmit & onComplete back there
- pass a part of state there (could we group these params together in order to pass say, `state.formData` ?)